File: WebApp/indoor-localization/src/data-access/mock-repositories/MockAssetRepository.ts
import { Asset } from "../../entities/Asset";
import { IAssetRepository } from "../repository-interfaces/IAssetRepository";

export class MockAssetRepository implements IAssetRepository {
    private assets: Asset[] = [];

    async Get(id: number): Promise<Asset | null> {
        return this.assets.find(asset => asset.id === id) || null;
    }

    async GetAll(): Promise<Asset[]> {
        return this.assets;
    }

    async Add(asset: Asset): Promise<boolean> {
        this.assets.push(asset);
        return true;
    }

    async Delete(id: number): Promise<boolean> {
        const index = this.assets.findIndex(asset => asset.id === id);
        if (index !== -1) {
            this.assets.splice(index, 1);
            return true;
        }
        return false;
    }

    async Update(updatedAsset: Asset): Promise<boolean> {
        const index = this.assets.findIndex(asset => asset.id === updatedAsset.id);
        if (index !== -1) {
            this.assets[index] = updatedAsset;
            return true;
        }
        return false;
    }
}

File: Android/indoor_localization/lib/domain/mock-repositories/mock_asset_repository.dart
import 'package:indoor_localization/domain/entities/asset.dart';
import 'package:indoor_localization/domain/repository-interfaces/asset_repository.dart';

class MockAssetRepository implements IAssetRepository {
  final List<Asset> _assets = [];

  @override
  Future<Asset?> get(int id) async {
    return _assets.firstWhere((asset) => asset.id == id);
  }

  @override
  Future<List<Asset>> getAll() async {
    return List.unmodifiable(_assets);
  }

  @override
  bool add(Asset asset) {
    return false;
  }

  @override
  bool delete(int id) {
    return false;
  }

  @override
  bool update(Asset updatedAsset)  {
    return false;
  }
}
